cli/command/service/opts.go

```diff
@@ -554,18 +555,45 @@ func (options *serviceOptions) ToServiceMode() (swarm.ServiceMode, error) {
 	switch options.mode {
 	case "global":
 		if options.replicas.Value() != nil {
-			return serviceMode, errors.Errorf("replicas can only be used with replicated mode")
+			return serviceMode, errors.Errorf("replicas can only be used with replicated or replicated-job mode")
 		}
 
 		if options.maxReplicas > 0 {
-			return serviceMode, errors.New("replicas-max-per-node can only be used with replicated mode")
+			return serviceMode, errors.New("replicas-max-per-node can only be used with replicated or replicated-job mode")
+		}
+		if options.maxConcurrent.Value() != nil {
+			return serviceMode, errors.New("max-concurrent can only be used with replicated-job mode")
 		}
 
 		serviceMode.Global = &swarm.GlobalService{}
 	case "replicated":
+		if options.maxConcurrent.Value() != nil {
+			return serviceMode, errors.New("max-concurrent can only be used with replicated-job mode")
+		}
+
 		serviceMode.Replicated = &swarm.ReplicatedService{
 			Replicas: options.replicas.Value(),
 		}
+	case "replicated-job":
+		concurrent := options.maxConcurrent.Value()
+		if concurrent == nil {
+			concurrent = options.replicas.Value()
+		}
+		serviceMode.ReplicatedJob = &swarm.ReplicatedJob{
+			MaxConcurrent:    concurrent,
```

Comment:
What happens server side if `MaxConcurrent` is not set? I wonder if this logic shouldn't be server side? 🤔

Reply:
Per the Docker API, `MaxConcurrent` will default to `1` if not set, and `TotalCompletions` will default to the value of `MaxConcurrent`. I took the "whatever sounded good at the time" approach to API design.

For the CLI, because the UI we're shooting for is a bit different, it felt more intuitive to have the user specify `replicas` like they're used to, and then allow them to separately set `max-concurrent`. Additionally, it seemed that in the absence of an explicit `max-concurrent` setting, the user probably wants everything to run at once, not be staggered. This is _not_ the case with the API because the API doesn't share overlapping fields the way the CLI shares overlapping flags.